lib/packwerk/parse_run.rb

```diff
@@ -83,18 +99,22 @@ def find_offenses(show_errors: false)
       offense_collection
     end
 
-    def serial_find_offenses
-      all_offenses = T.let([], T.untyped)
-      @files.each do |path|
-        offenses = yield path
-        all_offenses.concat(offenses)
+    sig { params(block: ProcessFileProc).returns(T::Array[Offense]) }
+    def serial_find_offenses(&block)
+      all_offenses = T.let([], T::Array[Offense])
+      begin
```

Comment:
I think it is better to remove this `begin/end`. Just let the method rescue the exception in the entire body

Reply:
This one is needed because Sorbet doesn't know that `all_offenses = ...` won't throw an error. So having the `begin` after that declaration let's Sorbet know that it's an array, not a nilable array:

![Uploading Screen Shot 2022-02-17 at 10.06.21 AM.png…]()